### ISSUE TITLE
Dropdown accessibility improvements

### ISSUE THREAD

**Issue by paulcpederson** [Esri/calcite-web#989]
- Ensure arrow keys move focus between menu items
- When tabbing away from the menu to the next focusable element on the page, the menu should be closed/collapsed.
- Each menu item needs to have `role=menuitem`.

**Comment by paulcpederson** [comment]
Will be part of v1.2.0
